rust/gsc/src/gsc/main.rs

```diff
@@ -15,7 +16,19 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     // connect to our db
     let connection = sea_orm::Database::connect("postgres://127.0.0.1/test").await?;
-    Migrator::up(&connection, None).await?;
+    let pending_migrations = Migrator::get_pending_migrations(&connection).await?;
+    if pending_migrations.len() != 0 {
+        let err = Error::new(
+            ErrorKind::Other,
+            format!(
+                "This gsc version expects {:?} additional migrations to be applied",
+                pending_migrations.len()
+            ),
+        );
+        tracing::error! {%err, "unperformed migrations, please apply these migrations before starting gsc"};
+        Err(err)?;
+    }
+    //Migrator::up(&connection, None).await?;
```

Comment:
Oh didn't quite realize it but I also added this feature. Please just consider this an extra and don't make me split it out lol.